[libunwind] new port

Melnytskyi opened the pull request:
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

Melnytskyi committed:
Initial support of libunwind

Melnytskyi committed:
Fix copyright install

Melnytskyi committed:
./vcpkg x-add-version --all

Melnytskyi committed:
Update git-tree

Melnytskyi commented:
@microsoft-github-policy-service agree

jimwang118 commented on ports/libunwind/portfile.cmake:
```diff
@@ -0,0 +1,19 @@
+
+
+file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")
```
A blank line needs to be added at the end of the file.

jimwang118 commented on ports/libunwind/portfile.cmake:
```diff
@@ -0,0 +1,19 @@
+
+vcpkg_from_git(
+    OUT_SOURCE_PATH SOURCE_PATH
+    URL "https://github.com/libunwind/libunwind.git"
+    REF 24947191d61dda869e039e0414fe97e9f594acd5
+    HEAD_REF "v1.7-stable"
```
```suggestion
    REPO libunwind/libunwind
    REF “v${VERSION}”
    HEAD_REF master
```

Melnytskyi replied:
`vcpkg_from_git` accepts full url to git
also, main branch has more commits, so i used stable branch(they keep versions in branches)

dg0yt replied:
You should use `vcpkg_from_github` instead.

Melnytskyi replied:
`vcpkg_from_github` download archives, libunwind repo not use archives feature instead they keep version in branch

dg0yt replied:
You misunderstand what `vcpkg_from_github`, or GH, is doing. The archives are created on the fly. It doesn't matter if REF is a commit, tag or branch.

dg0yt replied:
... and they are using regular git tags for releases.

jimwang118 commented on ports/libunwind/vcpkg.json:
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "libunwind",
+  "version": "1.7",
```
```suggestion
  "version": "1.7.2",
```

jimwang118 commented on ports/libunwind/vcpkg.json:
```diff
@@ -0,0 +1,18 @@
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    },
+    {
+      "name": "vcpkg-cmake-config",
+      "host": true
+    }
```
```suggestion
    }
```

jimwang118 commented:
Note: I will be converting your PR to draft status. The above suggested changes are only recommendations. If you are willing to adopt them, please click "ready for review" after making the modifications. If you do not wish to make any changes, please click "ready for review" directly. That way, I can be aware that you've responded since you can't modify the tags.

Melnytskyi committed:
Resolve some change requests

 - A blank line needs to be added at the end of the file.
 - Fix the version number in the vcpkg.json and usage files.

Melnytskyi committed:
Fix version

dg0yt commented on ports/libunwind/usage:
```diff
@@ -0,0 +1,6 @@
+The package libunwind can be imported via CMake FindPkgConfig module:
+
+    find_package(PkgConfig)
+    pkg_check_modules(libunwind REQUIRED IMPORTED_TARGET GLOBAL libunwind>=1.7.2)
```
```suggestion
    pkg_check_modules(libunwind REQUIRED IMPORTED_TARGET GLOBAL libunwind)
```

We don't want to update this file every time, and we don't know what the user actually needs.

dg0yt commented on ports/libunwind/portfile.cmake:
```diff
@@ -0,0 +1,19 @@
+vcpkg_minimum_required(VERSION 2022-10-12)
+
```
```suggestion
```

dg0yt commented on ports/libunwind/portfile.cmake:
```diff
@@ -0,0 +1,19 @@
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+)
+vcpkg_build_make(ENABLE_INSTALL)
```
```suggestion
vcpkg_install_make()
```

Melnytskyi committed:
Use vcpkg_from_github instead vcpkg_from_git

Melnytskyi committed:
Fix usage and portfile.cmake files

Melnytskyi committed:
Fix git-tree

dg0yt commented on ports/libunwind/portfile.cmake:
```diff
@@ -0,0 +1,18 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO "libunwind/libunwind"
+    REF v1.7.2
```
```suggestion
    REF "v${VERSION}"
```

dg0yt commented on ports/libunwind/portfile.cmake:
```diff
@@ -0,0 +1,18 @@
+)
+
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
```
```suggestion
    SOURCE_PATH "${SOURCE_PATH}"
```

dg0yt reviewed (commented):
The current state of the libunwind project is unclear.
Is the official main repo savannah or Github?
Is 1.7.2 an official release?

Someone else must approve

Melnytskyi commented:
> The current state of the libunwind project is unclear. Is the official main repo savannah or Github? Is 1.7.2 an official release?
> 
> Someone else must approve

Official repo is Github, in savannah lastest update(1.5) refer to github

1.7.2 is in github releases

Melnytskyi committed:
Yet another refactor

Melnytskyi committed:
Update git-tree

dg0yt commented:
> Official repo is Github, in savannah lastest update(1.5) refer to github

Okay, so the move is part of the "latest news" from savannah. It is a pity that they weren't able to update all the outdated information on the webpage.

jimwang118 commented:
Usage test pass with following triplets:

```
x64-linux
```
